### ISSUE TITLE
network_id: '*' incompatible with network 'develop'

### ISSUE THREAD

**Issue by ErikAbele** [trufflesuite/truffle#1824]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

When explicitly defining a network named 'develop' (e.g. to be able to use `console --network develop` or `exec ... --network develop`) it is no longer possible to use network_id: '*' with this network. This was working fine up until 5.0.4 but stopped somewhere with a later version (I've upgraded to 5.0.7 directly so I am not exactly sure which version introduced this behaviour).

Now, to be able to use a 'develop' network, a numerical network_id has to be specified, e.g. '5777'.

While this is not a major issue per se, it points to some 'unclean' coding and probably shouldn't error out in the way it does (see below).

## Steps to Reproduce

Add a network named 'develop' to truffle-config.js and specify network_id: '*' with it:

```
	networks: {
		'develop': {
			host: '127.0.0.1',
			port: 9545,
			network_id: '*'
		}
	}
```
## Expected Behavior

I'd expect `truffle develop` to start without an error but it exits with the following:

## Actual Results

```
Error: Error: [number-to-bn] while converting number "*" to BN.js instance, error: invalid number value. Value must be an integer, hex string, BN or BigNumber instance. Note, decimals are not supported. Given value: "*"
    at toBN (/usr/local/lib/node_modules/truffle/build/webpack:/~/web3-utils/src/utils.js:64:1)
    at Method.hexToNumber [as outputFormatter] (/usr/local/lib/node_modules/truffle/build/webpack:/~/web3-utils/src/utils.js:234:1)
    at Method.formatOutput (/usr/local/lib/node_modules/truffle/build/webpack:/~/web3-eth/~/web3-core-method/src/index.js:163:1)
    at sendTxCallback (/usr/local/lib/node_modules/truffle/build/webpack:/~/web3-eth/~/web3-core-method/src/index.js:473:1)
    at /usr/local/lib/node_modules/truffle/build/webpack:/~/web3-eth/~/web3-core-requestmanager/src/index.js:147:1
    at /usr/local/lib/node_modules/truffle/build/webpack:/packages/truffle-provider/wrapper.js:112:1
    at XMLHttpRequest.request.onreadystatechange (/usr/local/lib/node_modules/truffle/build/webpack:/~/web3/~/web3-providers-http/src/index.js:96:1)
    at XMLHttpRequestEventTarget.dispatchEvent (/usr/local/lib/node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request-event-target.js:34:1)
    at XMLHttpRequest._setReadyState (/usr/local/lib/node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:208:1)
    at XMLHttpRequest._onHttpResponseEnd (/usr/local/lib/node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:318:1)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/truffle/build/webpack:/~/xhr2-cookies/dist/xml-http-request.js:289:47)
    at IncomingMessage.emit (events.js:202:15)
    at IncomingMessage.EventEmitter.emit (domain.js:446:20)
    at endReadableNT (_stream_readable.js:1132:12)
    at processTicksAndRejections (internal/process/next_tick.js:76:17)
```

## Environment

* Operating System: Mac OS X 10.14.3
* Ethereum client: n/a
* Truffle version (`truffle version`): 5.0.7 and later
* node version (`node --version`): 11.11.0
* npm version (`npm --version`): 6.9.0

**Comment by CruzMolina** [comment]
Hey @ErikAbele ! Good catch. PR incoming to handle non-numericals.

**Comment by CruzMolina** [comment]
Closing as this has been merged via #1825 & released in truffle `5.0.9` 🎉 

Thanks @ErikAbele !

**Comment by ErikAbele** [comment]
Fix confirmed – thank you, @CruzMolina !
